File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.java
class Solution {
    public int minimumSumSubarray(List<Integer> nums, int l, int r) {
        int minSum = Integer.MAX_VALUE;
        boolean found = false;
        int n = nums.size();

        for (int i = 0; i <= n - l; i++) {
            for (int j = l; j <= r && i + j <= n; j++) {
                int sum = 0;
                for (int k = i; k < i + j; k++) {
                    sum += nums.get(k);
                }
                if (sum > 0) {
                    minSum = Math.min(minSum, sum);
                    found = true;
                }
            }
        }
        return found ? minSum : -1;
    }
}

File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.py
class Solution:
    def minimumSumSubarray(self, nums: List[int], l: int, r: int) -> int:

        mn = inf
        acc = list(accumulate(nums, initial=0))

        for left in range(len(acc) - l):
            for rght in range(left + l, min(left + r + 1, len(acc))):

                sm = acc[rght] - acc[left]
                if 0 < sm < mn:
                    mn = sm

        return -1 if mn == inf else mn